receipt_upload/receipt_upload/line_items/reconstructor.py

```
@@ -261,6 +266,71 @@ def cy(k: Tuple[int, int]) -> float:
+        key = (lab.line_id, lab.word_id)
+        val = val_by_key.get(key)
+        if val is None:
+            continue
+        by_value.setdefault(val, []).append(lab)
+
+    redundant: List[ReceiptWordLabel] = []
+    for _val, labs in by_value.items():
+        if len(labs) < 2:
+            continue
+        # Keep the lowest-on-receipt copy (smallest cy); invalidate the rest.
+        labs.sort(key=lambda l: pos.get((l.line_id, l.word_id), 0.0))
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)